src/vm_pages.rs

```diff
@@ -281,15 +283,14 @@ impl VmRegionList {
 
 /// Wrapper for a `PageTableMapper` created from the page table of `VmPages`. Measures pages as
 /// they are inserted, if necessary.
-pub struct VmPagesMapper<'a, T: GuestStagePageTable, S> {
+pub struct VmPagesMapper<'a, T: GuestStagePageTable> {
     mapper: PageTableMapper<'a, T>,
-    vm_pages: &'a VmPages<T, S>,
 }
 
-impl<'a, T: GuestStagePageTable, S> VmPagesMapper<'a, T, S> {
```

Comment:
Why remove the state type parameter (`S`)? It was there to enforce (via the type system) that we didn't add measured pages to a running VM.

Reply:
Sorry I missed that. I added the state type parameter back, through a phantom field.